src/rws_interface.cpp

```diff
@@ -117,6 +117,111 @@ std::string RWSInterface::getIOSignal(const std::string& iosignal)
   return result;
 }
 
+bool RWSInterface::getMechanicalUnitStaticInfo(const std::string& mechunit, MechanicalUnitStaticInfo* p_static_info)
+{
+  bool result = false;
+
+  if (p_static_info)
+  {
+    RWSClient::RWSResult rws_result = rws_client_.getMechanicalUnitStaticInfo(mechunit);
+
+    if(rws_result.success)
+    {
+      p_static_info->task_name = xmlFindTextContent(rws_result.p_xml_document, XMLAttribute("class", "task-name"));
+      p_static_info->is_integrated_unit = xmlFindTextContent(rws_result.p_xml_document,
+                                                             XMLAttribute("class", "is-integrated-unit"));
+      p_static_info->has_integrated_unit = xmlFindTextContent(rws_result.p_xml_document,
+                                                              XMLAttribute("class", "has-integrated-unit"));
+
+      std::string type = xmlFindTextContent(rws_result.p_xml_document, XMLAttribute("class", "type"));
+      if(type == "None")
+      {
+        p_static_info->type = NONE;
+      }
+      else if(type == "TCPRobot")
+      {
+        p_static_info->type = TCP_ROBOT;
+      }
+      else if(type == "Robot")
+      {
+        p_static_info->type = ROBOT;
+      }
+      else if(type == "Single")
+      {
+        p_static_info->type = SINGLE;
+      }
+      else
+      {
+        p_static_info->type = UNDEFINED;
+      }
+
+      std::stringstream axes(xmlFindTextContent(rws_result.p_xml_document, XMLAttribute("class", "axes")));
+      axes >> p_static_info->axes;
+
+      std::stringstream axes_total(xmlFindTextContent(rws_result.p_xml_document, XMLAttribute("class", "axes-total")));
+      axes_total >> p_static_info->axes_total;
+
+      result = true;
+    }
+  }
+
+  return result;
+}
+
+bool RWSInterface::getMechanicalUnitDynamicInfo(const std::string& mechunit, MechanicalUnitDynamicInfo* p_dynamic_info)
+{
+  bool result = false;
+
+  if (p_dynamic_info)
```

Comment:
See earlier comment about failing early.

Reply:
Should not be needed anymore since I changed to passing references instead of raw pointers.